### PR TITLE
Fix instance metadata service URL.

### PR DESCRIPTION
**Pull request by brendandburns**
Small fix @colemickens @jackfrancis

### CODE REVIEW

File: pkg/cloudprovider/providers/azure/azure_instance_metadata.go [hunk 1]
```diff
@@ -23,14 +23,26 @@ import (
...
 // This is just for tests injection
-var metadataURL = "http://169.254.169.254/metadata"
+const metaURL = "http://169.254.169.254/metadata/"
+
+var metadataURL = metaURL
 
 // SetMetadataURLForTesting is used to modify the URL used for
 // accessing the metadata server. Should only be used for testing!
 func SetMetadataURLForTesting(url string) {
 	metadataURL = url
```
Comment (colemickens) [inline pkg/cloudprovider/providers/azure/azure_instance_metadata.go]: Is this just for package local testing? Does it need to be public?

File: pkg/cloudprovider/providers/azure/azure_instance_metadata.go [hunk 2]
```diff
@@ -23,14 +23,26 @@ import (
...
 }
 
+// ResetMetadataURLForTesting resets the metadata service URL to the default.
+func ResetMetadataURLForTesting() {
```
Comment (colemickens) [inline pkg/cloudprovider/providers/azure/azure_instance_metadata.go]: Same comment about unexporting.

File: pkg/cloudprovider/providers/azure/azure_instance_metadata.go (hunk 1 above)
Comment (brendandburns) [inline pkg/cloudprovider/providers/azure/azure_instance_metadata.go]: Done.

File: pkg/cloudprovider/providers/azure/azure_instance_metadata.go (hunk 2 above)
Comment (brendandburns) [inline pkg/cloudprovider/providers/azure/azure_instance_metadata.go]: Done.

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Pull request merged**